### ISSUE TITLE
ImageInfo.FromStream only works on completely loaded images 

### ISSUE THREAD

**Issue by cs-util** [StbSharp/StbImageLib#3]
I am currently testing if its possible to run ImageInfo.FromStream on a stream that does not contain the full image but just the first bytes (so eg the first 5kb), but no matter how many bytes I put in the stream it always complains that it only can load the ImageInfo if the full stream is loaded:
```
    Error: System.Exception : EOF
    IOUtils.stbi__get8(Stream s) line 18
    JpgDecoder.stbi__process_marker(Int32 m) line 921
    JpgDecoder.stbi__decode_jpeg_header(Int32 scan) line 1219
    JpgDecoder.Info(Stream stream) line 1649
```

the use case is that the stream comes from a server and I don't want to download the full image, I just want to load the beginning of it so that I can check its pixel dimensions 

Any thoughts on how to make this work?

**Comment by rds1983** [comment]
Can you share the image?
As I've written a simple unit test: https://github.com/StbSharp/SafeStbImageSharp/blob/master/tests/SafeStbImageSharp.Tests/Tests.cs

It reads first 2k bytes from a jpg and runs ImageInfo on that data. And it works fine.

**Comment by cs-util** [comment]
This png for example makes problems when used with ImageInfo.FromStream: https://i.imgur.com/J7dAdPl.png

**Comment by rds1983** [comment]
Yup, reproduced. I even figured out what the issue was - it was bad idea to throw an exception in the stbi__err. Thanks for the report. I'll fix it soon.

**Comment by cs-util** [comment]
Great, thanks for looking into it so quickly 🙏

**Comment by rds1983** [comment]
Fixed in the version 2.22.4
